src/fsharp/LexFilter.fs

```diff
@@ -1099,14 +1099,14 @@ type LexFilterImpl (lightSyntaxStatus:LightSyntaxStatus, compilingFsLib, lexer,
             // span of inserted token lasts from the col + 1 of the prev token 
             // to the beginning of current token
             let lastTokenPos = 
-                let pos = tokenTup.LastTokenPos.Y
+                let pos = tokenTup.LastTokenPos
                 pos.ShiftColumnBy 1
-            returnToken (lexbufStateForInsertedDummyTokens (lastTokenPos, tokenTup.LexbufState.StartPos)) tok
+            returnDummyToken lastTokenPos tokenTup.LexbufState.StartPos tok
 
         let insertToken tok = 
             delayToken tokenTup
             if debug then dprintf "inserting %+A\n" tok
-            returnToken (lexbufStateForInsertedDummyTokens (startPosOfTokenTup tokenTup, tokenTup.LexbufState.EndPos)) tok
+            returnDummyToken (startPosOfTokenTup tokenTup) tokenTup.LexbufState.EndPos tok
```

Comment:
This cleans up the code nicely.

Reply:
We were allocating a lexBufState just to allocate the lexbuffer positions, made no sense, this seemed tidier